java/src/com/ibm/streamsx/topology/TStream.java

```diff
@@ -89,6 +90,41 @@
      * @return Filtered stream
      */
     TStream<T> filter(Predicate<T> filter);
+    
+    /**
+     * Split a stream into {@code n} streams as specified by {@code splitter}.
+     * 
+     * <P>
+     * For each tuple on the stream, {@code splitter.applyAsInt(tuple)} is called.
+     * The return value {@code r} determines the destination stream:
+     * <pre>
+     * if r < 0 the tuple is discarded
+     * else it is sent to the stream at position (r % n) in the returned array.
+     * </pre>
+     * </P>
+     * 
+     * <P>
+     * Example of round robin splitting a stream of {@code String} into 5 channels.
```

Comment:
With respect to parallel(..., ToIntFunction()), are you thinking a use case might be something like:
```java
TStream<String> names = getTStreamOfNames();

// Partition based on whether name begins with "A"
TStream<String>  partitionWithA = names.parallel(2, new Function<String, Integer>(){
    @Override
    public Integer apply(String name){
        if(name.startsWith("A")){
            return 0;
        } 
        else{
            return 1;
        }
    }});

// Partition based on whether name begins with "B"
TStream<String>  partitionWithB = names.parallel(2, new Function<String, Integer>(){
    @Override
    public Integer apply(String name){
        if(name.startsWith("B")){
            return 0;
        } 
        else{
            return 1;
        }
    }});
...
```

Where in the above case `names`, the same logical tuple stream, is partitioned in two different ways without creating any kind of intermediate transformation.